src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -15868,11 +15868,54 @@ public static Vector512<T> Invoke(Vector512<T> x)
         internal readonly struct AcoshOperator<T> : IUnaryOperator<T, T>
             where T : IHyperbolicFunctions<T>
         {
-            public static bool Vectorizable => false; // TODO: Vectorize
+            public static bool Vectorizable => typeof(T) == typeof(float) || typeof(T) == typeof(double);
+
             public static T Invoke(T x) => T.Acosh(x);
-            public static Vector128<T> Invoke(Vector128<T> x) => throw new NotSupportedException();
-            public static Vector256<T> Invoke(Vector256<T> x) => throw new NotSupportedException();
-            public static Vector512<T> Invoke(Vector512<T> x) => throw new NotSupportedException();
+
+            public static Vector128<T> Invoke(Vector128<T> x)
+            {
+                if (typeof(T) == typeof(float))
+                {
+                    Vector128<float> f = x.AsSingle();
+                    return LogOperator<float>.Invoke(f + Vector128.Sqrt(MultiplyAddEstimateOperator<float>.Invoke(f, f, Vector128.Create(-1f)))).As<float, T>();
```

Comment:
Isn't it supposed to be `log(f + sqrt(f - 1) * sqrt(f + 1))`?

Reply:
Doing `sqrt(f * f - 1)` only works assuming `f` is positive, which we can't make here without doing `abs` and `copysign`

Reply:
> Isn't it supposed to be log(f + sqrt(f - 1) * sqrt(f + 1))?

Why? Isn't the formula `acosh(x) = ln(x + sqrt(x^2 - 1))`?

> we need to ensure that invalid inputs still return NaN and that acos(1) returns exactly +0

It does, e.g. this:
```C#
using System.Numerics.Tensors;

float[] values = new float[] { 0f, 1f, 2f, 3f };
float[] dest = new float[values.Length];
TensorPrimitives.Acosh<float>(values, dest);

Console.WriteLine(float.IsNaN(dest[0]));
Console.WriteLine(BitConverter.SingleToUInt32Bits(dest[1]));
Console.WriteLine(BitConverter.SingleToUInt32Bits(dest[2]));
Console.WriteLine(BitConverter.SingleToUInt32Bits(dest[3]));
```
outputs:
```
True
0
1068012052
1071751603
```

> Doing sqrt(f * f - 1) only works assuming f is positive

Why does it assume that?